pilates/activitysim/preprocessor.py

```diff
@@ -697,22 +753,46 @@ def _create_land_use_table(
 
 
 def _get_zones_table(
-        datastore, region, asim_zone_id_col='TAZ',
-        default_zone_id_col='zone_id'):
+        datastore, region, state_fips, county_codes, zonification, 
+        asim_zone_id_col='TAZ', default_zone_id_col='zone_id'):
+    
     zone_key = '/zone_geoms'
     if zone_key in datastore.keys():
         logger.info("Loading zone geometries from .h5 datastore!")
         zones = datastore[zone_key]
-        if zones.index.name != asim_zone_id_col:
-            if asim_zone_id_col in zones.columns:
-                zones.set_index(asim_zone_id_col, inplace=True)
-            elif zones.index.name == default_zone_id_col:
-                zones.index.name = asim_zone_id_col
-            elif asim_zone_id_col not in zones.columns:
-                zones.rename(columns={default_zone_id_col: asim_zone_id_col})
-            else:
+        
+        if zones.index.name == default_zone_id_col:
+            pass
+        
+        elif zones.index.name in ['TAZ','GEOID']: 
+            zones.index.rename(default_zone_id_col, inplace = True)
+            
+        elif ('TAZ' in zones.columns) or ('GEOID' in zones.columns): 
+            replace_dict = {'TAZ': default_zone_id_col,'GEOID': default_zone_id_col}
+            try: 
+                zones.rename(columns = replace_dict, inplace = True)
+                zones.set_index(default_zone_id_col, inplace = True)
+            except ValueError:
                 logger.error(
+                    "Zones table has both TAZ and GEOID as zone_id!")
+        else:
+            logger.error(
                     "Not sure what column in the zones table is the zone ID!")
+            
+#         if zones.index.name != asim_zone_id_col:
+#             if asim_zone_id_col in zones.columns:
+#                 zones.set_index(asim_zone_id_col, inplace=True)
+#             elif zones.index.name == default_zone_id_col:
+#                 zones.index.name = asim_zone_id_col
+#             elif asim_zone_id_col not in zones.columns:
+#                 zones.rename(columns={default_zone_id_col: asim_zone_id_col})
+#             elif zones.index.name != 'GEOID':
+                
+#             elif 'GEOID' in zones.columns:
+                
+#             else:
+#                 logger.error(
+#                     "Not sure what column in the zones table is the zone ID!")
```

Comment:
If I'm understanding this correctly, these changes are mostly to make it so that activitysim will use "zone_id" instead of "TAZ" as the zone column, right? Because TAZ is a specific thing and if our zones are actually block groups we don't want to call them TAZs? Makes sense. But isn't the core ActivitySim code still going to be looking for columns names "TAZ" rather than "zone_id"? Won't we have to update the core ActivitySim code as well to reflect this change?

Maybe it would be easier to keep the column named "zone_id" in the .h5 tables, but leave them named "TAZ" in the ActivitySim tables (e.g. land_use.csv)?